src/connection/query.ts

```diff
@@ -0,0 +1,104 @@
+import { SQLJob } from "./sqlJob";
+import { CLCommandResult, JobLogEntry, QueryResult, ServerResponse } from "./types";
+export enum QueryState {
+  NOT_YET_RUN = 1,
+  RUN_MORE_DATA_AVAILABLE = 2,
+  RUN_DONE = 3,
+  ERROR = 4
+}
+export class Query<T> {
+
+  private static id_ctr: number = 43;

```

Comment:
This could have a better name.

Reply:
resolved at https://github.com/halcyon-tech/vscode-db2i/pull/23/commits/ba523bac2febfba9fdaa67d613a77e721b212df3